### PR TITLE
Support DynamoDb Client

### PR DESCRIPTION
**Pull request by yaohua-boey**

### CODE REVIEW

**Review by SaxJ — APPROVED** [review]

**Review by tspencer244 — APPROVED** [review]

**Comment by nathanblogs** [comment]
Potentially a breaking change shouldn't we bump the version ?

I think this will change the credential order which will result in a faster autneication https://docs.aws.amazon.com/sdk-for-php/v3/developer-guide/guide_configuration.html#credentials

File: tests/AdapterTest.php [hunk 1]
```diff
@@ -34,7 +34,7 @@ protected function getEnvironmentSetUp($app)
                 'Token'           => null,
                 'Expiration'      => "@{$expire}",
             ];
-            return promise_for(new Response(200, [], stream_for(json_encode($f))));
+            return promise_for(new Response(200, [], \GuzzleHttp\Psr7\Utils::streamFor(json_encode($f))));
```
Comment (brendan-markham-he) [inline tests/AdapterTest.php]: Is the `stream_for` above no longer used then? I'd swap the `use` for this new class.

**Review by brendan-markham-he — APPROVED** [review]

**Comment by tspencer244** [comment]
> Potentially a breaking change shouldn't we bump the version ?
> 
> I think this will change the credential order which will result in a faster autneication https://docs.aws.amazon.com/sdk-for-php/v3/developer-guide/guide_configuration.html#credentials

How would this be a breaking change?

Version bumps are done via Git tagging/Packagist.

File: tests/AdapterTest.php (hunk 1 above)
Comment (yaohua-boey) [reply tests/AdapterTest.php]: I've updated the imports for this test file

**Pull request merged**